test/withStyles_test.jsx

```diff
@@ -1,555 +1,619 @@
+        const StyledComponent = withStyles(stylesFn)(MockComponent);
+        const wrapper = mountWithProviders(<StyledComponent />, [
+          <DirectionProvider direction="ltr" />,
+        ]);
+        expect(stylesFn.calledWith(testTheme)).to.equal(true);
+        expect(wrapper.find(MockComponent).props().theme).to.equal(testTheme);
+      });
 
-      render((
-        <DirectionProvider direction={DIRECTIONS.LTR}>
-          <WrappedComponent />
-        </DirectionProvider>
-      ));
-      expect(resolveStub.callCount).to.equal(1);
+      it('uses the interface registered with the singleton API', () => {
+        const MockComponent = ({ styles, css }) => <div {...css(styles.primary)} />;
+        MockComponent.propTypes = { ...withStylesPropTypes };
+        const StyledComponent = withStyles(stylesFn)(MockComponent);
+        expect(testInterface.createLTR.callCount).to.equal(0);
+        expect(testInterface.resolveLTR.callCount).to.equal(0);
+        mountWithProviders(<StyledComponent />, [<DirectionProvider direction="ltr" />]);
+        expect(testInterface.createLTR.callCount).to.equal(1);
+        expect(testInterface.resolveLTR.callCount).to.equal(1);
+      });
     });
 
-    it('css calls resolveRTL method in RTL context', () => {
-      function MyComponent({ css }) {
-        return <div {...css({ color: 'red' })} />;
-      }
-      MyComponent.propTypes = {
-        ...withStylesPropTypes,
-      };
-
-      const WrappedComponent = withStyles(() => ({}))(MyComponent);
+    describe('in an RTL context', () => {
+      it('uses the theme registered with the singleton API', () => {
+        const MockComponent = ({ styles, css }) => <div {...css(styles.primary)} />;
+        MockComponent.propTypes = { ...withStylesPropTypes };
+        const StyledComponent = withStyles(stylesFn)(MockComponent);
+        const wrapper = mountWithProviders(<StyledComponent />, [
+          <DirectionProvider direction="rtl" />,
+        ]);
+        expect(stylesFn.calledWith(testTheme)).to.equal(true);
+        expect(wrapper.find(MockComponent).props().theme).to.equal(testTheme);
+      });
 
-      render((
-        <DirectionProvider direction={DIRECTIONS.RTL}>
-          <WrappedComponent />
-        </DirectionProvider>
-      ));
-      expect(resolveStub.callCount).to.equal(1);
+      it('uses the interface registered with the singleton API', () => {
+        const MockComponent = ({ styles, css }) => <div {...css(styles.primary)} />;
+        MockComponent.propTypes = { ...withStylesPropTypes };
+        const StyledComponent = withStyles(stylesFn)(MockComponent);
+        expect(testInterface.createRTL.callCount).to.equal(0);
+        expect(testInterface.resolveRTL.callCount).to.equal(0);
+        mountWithProviders(<StyledComponent />, [<DirectionProvider direction="rtl" />]);
+        expect(testInterface.createRTL.callCount).to.equal(1);
+        expect(testInterface.resolveRTL.callCount).to.equal(1);
+      });
```

Comment:
Thanks for adding/tweaking the test coverage! Makes it a lot easier for us to refactor and improve this code